packages/compiler-sfc/src/compileScript.ts

```
@@ -181,21 +189,37 @@
+    throw new Error(
+      `[@vue/compiler-sfc] <script> and <script setup> must have the same ` +
+        `language type.`,
+    )
+  }
+
+  if (scriptSetupLang && !isJSOrTS) {
     // do not process non js/ts script blocks
     return scriptSetup
   }
 
+  const ctx = new ScriptCompileContext(sfc, options)
```

Biome (2.1.2)

[error] Shouldn't redeclare 'ctx'. Consider to delete it or rename it.

'ctx' is defined here:


(lint/suspicious/noRedeclare)